### PR TITLE
Added baseline class specification

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Describes the approach for #1

### CODE REVIEW

**Comment by TwitchBronBron** [comment]
@georgejecook Can you please take a look at my class inheritance BrightscriptOutput, and make sure that will satisfy all of your concerns? I think I eliminated the need for a separate builder for classes by aliasing the super methods with their index (i.e. `super0`, `super1`, `super2`, etc...)

File: docs/classes.md [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    'specified private
+    instance.setSocialSecurityNumber = sub(value as string)
+        m.socialSecurityNumber = value
+    end sub
+    return instance
+end function
+```
+</details>
+
+You will get compile-time errors whenever you access private members of a class from outside the class. However, be aware that this is only a compile-time restriction. At runtime, all members are public.
```
Comment (georgejecook) [inline docs/classes.md:372]: what is the position on extended properties? I see no protected keyword (I think I have submitted an issue for that, if not I shall); but for now, do we need to be explicit about how you intend  private visibility to work with inheritance?

File: docs/classes.md [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+    instance.getAge = function()
+        return m.age
+    end function
+    return instance
+end function
+```
+</details>
+
+## Usage
+In order to use a class, you need to construct one. Based on our person class above, you can create a new person like this:
```
Comment (georgejecook) [inline docs/classes.md:414]: worth explicitly calling out the new `new` keyword?

File: docs/classes.md (hunk 1 above)
Comment (TwitchBronBron) [reply docs/classes.md:372]: The position is, they don't exist. A child class and super class will share all of the same properties. The compiler will add an error if a child and parent declare a property with the same name.

File: docs/classes.md (hunk 2 above)
Comment (TwitchBronBron) [reply docs/classes.md:414]: That's already explained in this section: https://github.com/rokucommunity/brighterscript/pull/40/files#diff-3031a75b5d15117a376b3869ba2a5b24R143

But I could probably add a little better documentation around it, or maybe declare that section higher up?

File: docs/classes.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+    instance = __Duck_Build()
+    instance.new(name)
+    return instance
+end function
+
+function __BabyDuck_Build()
+    instance = __Duck_Build()
+    instance.super1_move = instance.move
+    instance.move = sub(distanceInMeters as integer)
+        instance.super1_move(distanceInMeters)
```
[the remark this answers is not in the document]
Comment (TwitchBronBron) [reply docs/classes.md:118]: Oops, that's a typo. You're right, it needs to be `m`. I updated the docs in master. Thanks for catching this!

**Pull request merged**